### DISCUSSION TITLE
Unable to to Train even PointSetBC in TimePDE Problem

### DISCUSSION THREAD

**Discussion by ppandit95** [lululxvi/deepxde#1398] [Q&A]
Dear Dr. @lululxvi 

I was working with the inverse problem with spatially varying diffusivity in Fick's second law as 
`
def pde(x, y):
    dy_t = dde.grad.jacobian(y, x, i=0, j=1)
    dy_x = dde.grad.jacobian(y,x,i=0,j=0)
    dy_xx = dde.grad.jacobian(torch.exp(a*y + b*y*y + c*y*y*y+d0)*dy_x,x,i=0,j=0)
    return (dy_t-dy_xx)
' 
Followed by Homogeneous Neumann BCs as - 
'
def func(x):
    return np.zeros((len(x),1))
bc = dde.icbc.NeumannBC(geomtime, func, lambda _, on_boundary: on_boundary)
'
As well as a stepped profile as initial conditions as -
'
def step_function(x):
    comp=np.zeros((len(x),1))
    for i in range(len(x)):
        if i <= len(x)/2.0:
            comp[i] = 0.75 
        else:
            comp[i] = 0.25
    return comp

ic = dde.icbc.IC(geomtime, step_function, lambda _, on_initial: on_initial)
' 
Also, some of the observed data at final time step is added from a file . But the issue that I am facing is that my loss for initial conditions as well as loss for coomposition data at final time step isnt decreasing at all no matter for how many epoches I run the code.
Hopefully looking for a header to resolve this issue

Thanks in Advance
Pushkar
